tests/main/uc20-snap-recovery-encrypt/task.yaml

```diff
@@ -0,0 +1,120 @@
+summary: Integration tests for the snap-bootstrap binary
+
+# one system is enough, its a very specialized test for now
+systems: [ubuntu-19.10-64]
+
+debug: |
+    cat /proc/partitions
+
+restore: |
+    if [ -f loop.txt ]; then
+        losetup -d "$(cat loop.txt)"
+    fi
+
+prepare: |
+    echo "Create a fake block device image"
+    truncate --size=2GB fake.img
+
+    echo "Setup the image as a block device"
+    # without -P this test will not work, then /dev/loop1p? will be missing
+    losetup -fP fake.img
+    losetup -a |grep fake.img|cut -f1 -d: > loop.txt
+    LOOP="$(cat loop.txt)"
+
+    echo "Create an empty partition header"
+    echo "label: gpt" | sfdisk "$LOOP"
+
+    # XXX: update to the UC20 gadget when it's available
```

Comment:
Given that we have a uc20 gadget now, should we use that instead?

Reply:
This would be a good time to do it, I'll update accordingly.

Reply:
Using the uc20 gadget bring a scenario we didn't previously expect: the gadget compatibility check fails if we already created an encrypted partition. This won't affect our use case, but I'll add an XXX comment and fix in a follow-up if necessary.

Reply:
Hm, I wonder if we should expand the gadget capabilities somehow for this. Probably worth another XXX